pkg/engine/plugins.go

```diff
@@ -190,15 +192,11 @@ func gatherPluginsFromSnapshot(plugctx *plugin.Context, target *deploy.Target) (
 	return set, nil
 }
 
-// ensurePluginsAreInstalled inspects all plugins in the plugin set and, if any plugins are not currently installed,
+// EnsurePluginsAreInstalled inspects all plugins in the plugin set and, if any plugins are not currently installed,
 // uses the given backend client to install them. Installations are processed in parallel, though
 // ensurePluginsAreInstalled does not return until all installations are completed.
-func ensurePluginsAreInstalled(
-	ctx context.Context,
-	opts *deploymentOptions,
-	d diag.Sink,
-	plugins pluginSet,
-	projectPlugins []workspace.ProjectPlugin,
+func EnsurePluginsAreInstalled(ctx context.Context, opts *deploymentOptions, d diag.Sink, plugins PluginSet,
+	projectPlugins []workspace.ProjectPlugin, reinstall, explicitInstall bool,
```

Comment:
```suggestion
	projectPlugins []workspace.ProjectPlugin, reinstall, installMissing bool,
```

Maybe a better name?